### PR TITLE
CodeWhipserer completion integration tests

### PR DESCRIPTION
**Pull request by andrewyuq**
This change tests basic get completion feature
```
software.aws.toolkits.jetbrains.services.codewhisperer.CodeWhispererCodeScanIntegrationTest
  Test testCodeScanUnsupportedLanguage PASSED (11.9s)
  Test testCodeScanValidWithIssues PASSED (19.6s)
  Test testCodeScanValidWithNoIssues PASSED (18.7s)
  Test testCodeScanFileTooLarge PASSED
software.aws.toolkits.jetbrains.services.codewhisperer.CodeWhispererCodeScanJavaIntegrationTest
  Test testCodeScanJavaProjectNoBuild PASSED
software.aws.toolkits.jetbrains.services.codewhisperer.CodeWhispererCompletionIntegrationTest
  Test testInvokeCompletionUnsupportedLanguage PASSED
  Test testInvokeCompletionManualTrigger PASSED (8s)
  Test testInvokeCompletionAutoTrigger PASSED (6s)
SUCCESS: Executed 8 tests in 1m 8s
```
The only manual step here is to complete the browser login flow, the other parts will run automatically.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/it/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererIntegrationTestBase.kt [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        telemetryServiceSpy = spy(CodeWhispererTelemetryService.getInstance())
+        ApplicationManager.getApplication().replaceService(CodeWhispererTelemetryService::class.java, telemetryServiceSpy, disposableRule.disposable)
+
+        stateManager = CodeWhispererExplorerActionManager.getInstance()
+        stateManager.setHasAcceptedTermsOfService(true)
+        stateManager.setAutoEnabled(false)
+
+        popupManager = spy(CodeWhispererPopupManager.getInstance())
+        popupManager.reset()
+        doNothing().`when`(popupManager).showPopup(any(), any(), any(), any(), any())
```
Comment (rli) [inline jetbrains-core/it/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererIntegrationTestBase.kt:106]:
```suggestion
        doNothing().whenever(popupManager).showPopup(any(), any(), any(), any(), any())
```
?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanTestBase.kt [hunk 2]
```diff
@@ -39,6 +39,9 @@ import software.aws.toolkits.jetbrains.services.codewhisperer.credentials.CodeWh
 import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhispererExplorerActionManager
 import software.aws.toolkits.jetbrains.utils.isInstanceOf
 import software.aws.toolkits.jetbrains.utils.rules.CodeInsightTestFixtureRule
+import java.io.File
+import java.nio.file.Path
+import kotlin.io.path.relativeTo
```
Comment (Will-ShaoHua) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanTestBase.kt:44]: unused import?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/credentials/CodeWhispererClientAdaptor.kt [hunk 3]
```diff
@@ -68,6 +70,13 @@ interface CodeWhispererClientAdaptor : Disposable {
 class CodeWhispererClientAdaptorImpl(override val project: Project) : CodeWhispererClientAdaptor {
     private val mySigv4Client by lazy { createUnmanagedSigv4Client() }
 
+    // Will purge once sigV4 client for CW is no longer needed
+    private val KProperty0<*>.isLazyInitialized: Boolean
+        get() {
+            isAccessible = true
+            return (getDelegate() as Lazy<*>).isInitialized()
+        }
+
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/credentials/CodeWhispererClientAdaptor.kt:79]: just curious is this change required  to call `this::mySigv4Client.isLazyInitialized`

**Review by Will-ShaoHua — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/credentials/CodeWhispererClientAdaptor.kt (hunk 3 above)
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/credentials/CodeWhispererClientAdaptor.kt:79]: yes, and previously we have the risk of calling client.close() without it being initialized, hence this change.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/toolwindow/CodeWhispererCodeReferenceComponents.kt [hunk 4]
```diff
@@ -69,7 +68,7 @@ class CodeWhispererCodeReferenceComponents(private val project: Project) {
         repaint(project)
 
         // set the reference panel text different for SSO users vs AWS Builder ID / Accless users
-        ApplicationManager.getApplication().messageBus.connect().subscribe(
+        project.messageBus.connect().subscribe(
```
[the remark this answers is not in the document]
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/toolwindow/CodeWhispererCodeReferenceComponents.kt:71]: My understanding is that the messageBus belongs to the project so when the project gets disposed, the connection will also get released?

[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/toolwindow/CodeWhispererCodeReferenceComponents.kt:71]: ok that make sense

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/CodeWhispererExplorerActionManager.kt [hunk 5]
```diff
@@ -187,7 +187,7 @@ internal class CodeWhispererExploreActionState : BaseState() {
 }
 
 // TODO: Don't remove IsManualEnabled
```
Comment (saurabhajmera) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/CodeWhispererExplorerActionManager.kt:189]: do we still need this TODO, since IsManualEnabled is already part of the enum class? May be you should consider adding some context here to indicate why someone should not remove IsManualEnabled and remove the `TODO:` prefix

**Review by saurabhajmera — APPROVED** [review]
left a minor comment, approving to unblock

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/CodeWhispererExplorerActionManager.kt (hunk 5 above)
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/CodeWhispererExplorerActionManager.kt:189]: This is currently unused, but we need to make sure removing it is backward-compatiable with the existing schema for loading/saving states. This is worth to be addressed in a standalone PR.
